ivy/data_classes/factorized_tensor/tt_matrix.py

```diff
@@ -0,0 +1,282 @@
+
+        Re-assembles 'factors', which represent a tensor in TT-Matrix format
+        into the corresponding matrix
+
+        Parameters
+        ----------
+        factors
+            TT-Matrix factors (known as core)
+            of shape (rank_k, left_dim_k, right_dim_k, rank_{k+1})
+
+        Returns
+        -------
+        output_matrix
+            matrix whose TT-Matrix decomposition was given by 'factors'
+        """
+        in_shape = tuple(c.shape[1] for c in tt_matrix)
+        return ivy.reshape(
+            ivy.TTMatrix.tt_matrix_to_tensor(tt_matrix), (ivy.prod(in_shape), -1)
+        )
+
+    @staticmethod
+    def tt_matrix_to_tensor(tt_matrix):
+        """
+        Return the full tensor whose TT-Matrix decomposition is given by 'factors'.
+
+        Re-assembles 'factors', which represent a tensor in TT-Matrix format
+        into the corresponding full tensor
+
+        Parameters
+        ----------
+        factors
+            TT-Matrix factors (known as core)
+            of shape (rank_k, left_dim_k, right_dim_k, rank_{k+1})
+
+        Returns
+        -------
+        output_tensor
+            tensor whose TT-Matrix decomposition was given by 'factors'
+        """
+        _, in_shape, out_shape, _ = zip(*[ivy.shape(f) for f in tt_matrix])
+        ndim = len(in_shape)
+
+        full_shape = sum(zip(*(in_shape, out_shape)), ())
+        order = list(range(0, ndim * 2, 2)) + list(range(1, ndim * 2, 2))
+
+        for i, factor in enumerate(tt_matrix):
+            if not i:
+                res = factor
+            else:
+                res = ivy.tensordot(res, factor, ([-1], [0]))
+
+        # return ivy.transpose(ivy.reshape(res, full_shape), order)
+        return ivy.permute_dims(ivy.reshape(res, full_shape), order)
+
+    @staticmethod
+    def tt_matrix_to_vec(tt_matrix):
+        """
+        Return the tensor defined by its TT-Matrix format ('factors') into its
+        vectorized format.
+
+        Parameters
+        ----------
+        factors
+            TT factors
+
+        Returns
+        -------
+        1-D array
+            format of tensor defined by 'factors'
+        """
+        return ivy.reshape(ivy.TTMatrix.tt_matrix_to_tensor(tt_matrix), (-1,))
+
+    @staticmethod
+    def tt_matrix_to_unfolded(tt_matrix, mode):
+        """
+        Return the unfolding matrix of a tensor given in TT-Matrix format.
+
+        Reassembles a full tensor from 'factors' and returns its unfolding matrix
+        with mode given by 'mode'
+
+        Parameters
+        ----------
+        factors
+            TT-Matrix factors
+        mode
+            unfolding matrix to be computed along this mode
+
+        Returns
+        -------
+        2-D array
+            unfolding matrix to be computed along this mode
+        """
+        return ivy.unfold(ivy.TTMatrix.tt_matrix_to_tensor(tt_matrix), mode=mode)
+
+    @staticmethod
+    def validate_tt_matrix_rank(tensorized_shape, rank="same"):
```

Comment:
@hello-fri-end The assertion tests are failing because of this error while I try to call `ivy.TTMatrix.validate_tt_matrix_rank` in my implementation of `ivy.random_tt_matrix`:
```
'TTMatrix' has no attribute 'validate_tt_matrix_rank'
```
I don't know what's causing the bug